### ISSUE TITLE
Security review

### ISSUE THREAD

**Issue by DukeyToo** [perfectapi/ChromeKeePass#10]
Do a security review against the [OWASP top 10](https://www.owasp.org/index.php/Top_10_2013-Top_10) and make any changes necessary to minimize risk.

Document the review on the wiki.

**Comment by DukeyToo** [comment]
Security review complete, and appropriate changes made to the code.  https://github.com/perfectapi/ChromeKeePass/wiki/OWASP-Security-Review

**Comment by solikeumyeah** [comment]
I noticed you still have some open questions on that Wiki link. Not sure where to discuss them?

Regarding A6:
> The only unencrypted data the app stores are key files. How easy is it for someone with physical access to gain access to the key file?

I found my keyfile + kdbx inside this file:
`%LOCALAPPDATA%\Google\Chrome\User Data\Default\Local Extension Settings\lnfepbjehgokldcaljagbmchhnaaogpc/000003.log`

The permissions on this file were inherited from the user profile, and therefore not readable by other non-privileged users on the system. 

The user could add additional security by enabling Windows EFS encryption on the Local Extension Settings folder. This would, for example, make it more difficult to get the keyfile by extracting the disk physically from the machine (especially if combined with Windows Bitlocker), and would provide protection against accidentally sharing your Chrome profile (e.g. if the NTFS permissions become configured incorrectly, or if your backup drive is not secure). This may be a suitable recommendation for the Readme or Wiki.
